homeassistant/components/wilight/switch.py

```diff
@@ -0,0 +1,393 @@
+ATTR_TRIGGER_2_DESC = "trigger_2_description"
+ATTR_TRIGGER_3_DESC = "trigger_3_description"
+ATTR_TRIGGER_4_DESC = "trigger_4_description"
+
+# Attr of services data supported by the valve switch entities
+ATTR_TRIGGER = "trigger"
+ATTR_TRIGGER_INDEX = "trigger_index"
+
+# Service of features supported by the valve switch entities
+SERVICE_SET_WATERING_TIME = "set_watering_time"
+SERVICE_SET_PAUSE_TIME = "set_pause_time"
+SERVICE_SET_TRIGGER = "set_trigger"
+
+# Range of features supported by the valve switch entities
+RANGE_WATERING_TIME = 1800
+RANGE_PAUSE_TIME = 24
+RANGE_TRIGGER_INDEX = 4
+
+# Service call validation schemas
+VALID_WATERING_TIME = vol.All(
+    vol.Coerce(int), vol.Range(min=1, max=RANGE_WATERING_TIME)
+)
+VALID_PAUSE_TIME = vol.All(vol.Coerce(int), vol.Range(min=1, max=RANGE_PAUSE_TIME))
+VALID_TRIGGER = wl_trigger
+VALID_TRIGGER_INDEX = vol.All(
+    vol.Coerce(int), vol.Range(min=1, max=RANGE_TRIGGER_INDEX)
+)
+
+# Service schemas
+SERVICE_SCHEMA_WATERING_TIME = vol.Schema(
+    {
+        vol.Required(ATTR_ENTITY_ID): cv.entity_id,
+        vol.Required(ATTR_WATERING_TIME): VALID_WATERING_TIME,
+    }
+)
+SERVICE_SCHEMA_PAUSE_TIME = vol.Schema(
+    {
+        vol.Required(ATTR_ENTITY_ID): cv.entity_id,
+        vol.Required(ATTR_PAUSE_TIME): VALID_PAUSE_TIME,
+    }
+)
+SERVICE_SCHEMA_TRIGGER = vol.Schema(
+    {
+        vol.Required(ATTR_ENTITY_ID): cv.entity_id,
+        vol.Required(ATTR_TRIGGER_INDEX): VALID_TRIGGER_INDEX,
+        vol.Required(ATTR_TRIGGER): VALID_TRIGGER,
+    }
+)
+
+# Descriptions of the valve switch entities
+DESC_WATERING = "watering"
+DESC_PAUSE = "pause"
+
+# Icons of the valve switch entities
+ICON_WATERING = "mdi:water"
+ICON_PAUSE = "mdi:pause-circle-outline"
+
+
+def entities_from_discovered_wilight(api_device: Any) -> tuple[Any]:
+    """Parse configuration and add WiLight switch entities."""
+    entities: Any = []
+    for item in api_device.items:
+        if item["type"] == ITEM_SWITCH:
+            index = item["index"]
+            item_name = item["name"]
+            if item["sub_type"] == SWITCH_VALVE:
+                entities.append(WiLightValveSwitch(api_device, index, item_name))
+            elif item["sub_type"] == SWITCH_PAUSE_VALVE:
+                entities.append(WiLightValvePauseSwitch(api_device, index, item_name))
+
+    return entities
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up WiLight switches from a config entry."""
+    parent = hass.data[DOMAIN][entry.entry_id]
+
+    # Handle a discovered WiLight device.
+    entities = entities_from_discovered_wilight(parent.api)
+    async_add_entities(entities)
+
+    # Handle services for a discovered WiLight device.
+    async def set_watering_time(service: Any) -> None:
+        entity_id = service.data[ATTR_ENTITY_ID]
+
+        for entity in entities:
+            if entity.entity_id == entity_id:
```

Comment:
Reverse the condition, continue and outdent below

Reply:
Sorry but I didn't understand. Could you please explain it to me in another way or give an example?
Thanks.